src/connection.rs

```diff
@@ -21,11 +21,13 @@ impl Connection {
     fn init(dbus_manager: &Rc<DBusNetworkManager>, path: &str) -> Result<Self, String> {
         let settings = dbus_manager.get_connection_settings(path)?;
 
-        Ok(Connection {
-               dbus_manager: dbus_manager.clone(),
-               path: path.to_string(),
-               settings: settings,
-           })
+        Ok(
+            Connection {
```

Comment:
Yes, that particular part felt odd to me as well when I added it. I will check which option did that and what else it involved.